lib/mgmt_msg_native.h

```diff
@@ -237,19 +252,33 @@ _Static_assert(sizeof(struct mgmt_msg_tree_data) ==
 #define GET_DATA_FLAG_STATE	0x01	/* get only "config false" data */
 #define GET_DATA_FLAG_CONFIG	0x02	/* get only "config true" data */
 #define GET_DATA_FLAG_EXACT	0x04	/* get exact data node instead of the full tree */
+/*
+ * The following 2 bits control "with-defaults" behavior as defined in RFC 6243.
+ * Possible values are:
+ * 00 - explicit
+ * 01 - trim
+ * 10 - report-all-tagged
+ * 11 - report-all
+ */
+#define GET_DATA_FLAG_WD_EXPLICIT 0x00
+#define GET_DATA_FLAG_WD_TRIM	  0x08
+#define GET_DATA_FLAG_WD_ALL_TAG  0x10
+#define GET_DATA_FLAG_WD_ALL	  0x18
+#define GET_DATA_FLAG_WD_MASK	  0x18
```

Comment:
Same comment as above. We are tightly coupled to libyang. There's no reason not to use the libyang constants. In fact these values here appear to be different from the exact same symbolic constants in libyang which is even worse b/c now we have to have a mapping. Please use libyang constants and types instead of creating new aliases.

Reply:
The reasoning is explained in the previous comment. In this particular case I don't like libyang flags for an additional reason - they are mutually exclusive, but can be set simultaneously. It is possible set `WD_TRIM` (remove all defaults) and `WD_ALL` (print all defaults) at the same time. I don't even know what will happen, but this is not a good API.

Reply:
I think these flags values are incorrect _ALL here doesn't mean all settings, and a flag should be a bit not multiple bits. The three values of `EXPLICIT` (only present nodes), `TRIM` (only nodes different from default value) and `ALL` (all nodes present or not) are mutually exclusive; and that's just the way it is, but we can't (and shouldn't try) to use the flag values to enforce that here.

Using the libyang values works with our other flags, and eliminates the need for mapping. So here's my suggestion. For both sets of libyang mapped values, use the libyang values, write 2 mapping inline functions that do nothing but return the value (b/c the values are the same) and `_Static_assert` the fact that the values are indeed the same (using libyang symbols). The latter can be done anywhere (including a .c file) and just refer back to the mapping functions in a comment to keep the header cleaner I suppose.

Reply:
Right I would rename `ALL` to `ALL_NODES` and I would rename `ALL_TAG` to `TAG_ALL`, makes it a lot more clear what's going on IMO.

Reply:
We have 4 different values here - `TRIM`, `EXPLICIT`, `ALL` and `ALL_TAG`. So we just need two bits to encode them. That's why I programmed it like that. I agree that it's a bit misleading that it's called a `flag` when it actually represents a value. How about we keep the encoding as-is, but change the semantics of the `flags` field to a bitfield:
```
uint8_t state         : 1;
uint8_t config        : 1;
uint8_t exact         : 1;
uint8_t with_defaults : 2;
uint8_t rsvd          : 3;
```
I really don't want to use libyang values directly here for two resons:
- as I wrote above, with-default is four distinct values, so exposing them as flags makes an impression that they are not mutually exclusive and can be used simultaneously
- we'll need at least uint16_t to store libyang values directly, because the largest value is `0x80`, and it's just to store values that can be encoded as two bits

Regarding the rename, the current names reflect values from the RFC 6243: `trim`, `explicit`, `report-all`, `report-all-tagged`. I can rename `ALL` and `ALL_TAG` to `REPORT_ALL` and `REPORT_ALL_TAGGED` to make it even more similar. But I don't want to use some custom names here.

What do you think?

Reply:
Ok fine we have 4 distinct values in 2 bits, but please don't call them flags then. So, remove _FLAG_ from their symbolic names and put them in another uint8 field maybe called `withdef` or something -- we have 6 unused uint8s currently.

I don't care very much about the RFC names here b/c I don't think people are going to be coming from the RFC when they read the code, they are just going to be looking at the code, and importantly the RFC names are confusing sounding, 

If I read `REPORT_ALL_TAGGED` I'm going to think, rightly so, that this is going to report all nodes that are tagged. Instead what this value means is `Return all values and add tags to them` so `ALL_ADD_TAG` would be a much better name.

You can mention the RFC name in a comment on the #define for the symbol if you think there's some value there.

Reply:
Implemented using a new `defaults` field in the message. Definitions renamed according to the request.